osu.Game/Rulesets/Scoring/ScoreProcessor.cs

```diff
@@ -98,6 +98,16 @@ public partial class ScoreProcessor : JudgementProcessor
 
         private readonly Bindable<ScoreRank> rank = new Bindable<ScoreRank>(ScoreRank.X);
 
+        /// <remarks>
+        /// Using the minimum accuracy as the data source to calculate the minimum rank.
+        /// </remarks>
+        public readonly Bindable<ScoreRank> MinimumRank = new Bindable<ScoreRank>(ScoreRank.X);
+
+        /// <remarks>
+        /// Using the maximum accuracy as the data source to calculate the minimum rank.
+        /// </remarks>
+        public readonly Bindable<ScoreRank> MaximumRank = new Bindable<ScoreRank>(ScoreRank.X);
```

Comment:
I'm having very mixed feelings about this living in `ScoreProcessor`. Maximum and minimum accuracy I can live with, since it isn't possible to calculate that from outside, but the rank is derived entirely from the score and the hit statistics, so I'm not sure I want this to live here (rather than locally in the skinnable component).

Reply:
> I'm having very mixed feelings about this living in `ScoreProcessor`. Maximum and minimum accuracy I can live with, since it isn't possible to calculate that from outside, but the rank is derived entirely from the score and the hit statistics, so I'm not sure I want this to live here (rather than locally in the skinnable component).

From my perspective, I think it would be suitable for the minimum and maximum rank to be calculated within the `ScoreProcessor`. It appears that the logic for determining the minimum and maximum rank is shared across different rulesets, and there are two components (`LegacyRankDisplay` and `DefaultRankDisplay`) that require bind to these data.

Reply:
I probably wouldn't want `Accuracy` or `Score` or any of the other bindables to be publicly editable either. Not sure if there's a use for them to be.

I think I'm okay with this PR for the time being, but I'd like to see them exposed as `IBindable` instead to follow the shape of `Rank`. Reason being here is that mods have modified `Rank` directly in the past which is how we ended up with the current `IBindable`.

Reply:
> I probably wouldn't want `Accuracy` or `Score` or any of the other bindables to be publicly editable either. Not sure if there's a use for them to be.
> 
> I think I'm okay with this PR for the time being, but I'd like to see them exposed as `IBindable` instead to follow the shape of `Rank`. Reason being here is that mods have modified `Rank` directly in the past which is how we ended up with the current `IBindable`.

Implement the change at the lastest commit, the `MaximumRank` and `MinimumRank` are now `IBindable<ScoreRank>`